src/main/java/com/socialmediaraiser/twitterbot/impl/personalAnalyzer/DataArchiveHelper.java

```diff
@@ -78,7 +77,31 @@ private void countRetweetsOfTweet(TweetDTOv1 tweet, UserInteractions userInterac
         }
     }
 
-    public void countGivenRetweets(UserInteractions userInteractions){
+    // new
+
+    public Map<String, TweetInteraction> countRetweetsReceived(){
+
+        LOGGER.info("\ncounting retweets received (archive)...");
```

Comment:
You are using a very poor logger.
Try to use an async Logback or Log4J logger, and eventually wrap it with SLF4J to give it more abstraction.

Reply:
By the way, the `/n` should not be necessary using a decent logger.

Reply:
I believe @Logs from Lombok was enough, going to check that, thanks !
I added the `\n` because two be able to follow the progression of tasks in IDE console, I have some dirty lines like 
`System.out.print(".");\  hahaha. If you have a better idea to follow things in console, i'm opened :)

Reply:
You can just `LOGGER.trace()` using a decent logger.
Loggers are not for user, so printing a `.` is not really an information.
You could `LOGGER.trace("Found a tweet corresponding to XXX")`.
`TRACE` is the lowest level, so I think using your current logger it should be `LOGGER.lowest("Found a tweet corresponding to XXX")`.

Reply:
My need was more to have a progression bar in just one line in console to avoid being too flooded when I have 10K iterations but being able to see where the progression is.